query_optimizer/ExecutionGenerator.cpp

```diff
@@ -670,10 +670,22 @@ void ExecutionGenerator::convertHashJoin(const P::HashJoinPtr &physical_plan) {
   }
 
   // Create join hash table proto.
-  const QueryContext::join_hash_table_id join_hash_table_index =
-      query_context_proto_->join_hash_tables_size();
+  std::vector<QueryContext::join_hash_table_id> join_hash_table_indices;
+
   S::HashTable *hash_table_proto = query_context_proto_->add_join_hash_tables();
 
+  if (build_relation_info->relation->hasPartitionScheme()) {
+    const std::size_t num_partitions =
+        build_relation_info->relation->getPartitionScheme().getPartitionSchemeHeader().getNumPartitions();
+    join_hash_table_indices.resize(num_partitions);
+  } else {
+    join_hash_table_indices.resize(1);
+  }
+
+  for (int i = 0; i < join_hash_table_indices.size(); ++i) {
+    join_hash_table_indices[i] = query_context_proto_->join_hash_tables_size();
```

Comment:
If I understand correctly, with partition, we should create one hash table (proto) per partition. But here I believe it is a bug that we copy the only hash table for various partitions.